src/tensors/cpu/intgemm_interface.h

```diff
@@ -76,15 +76,21 @@ template<Type vtype>
 struct PrepareBNodeOp : public NaryNodeOp {
 float clipValue_;
 float quantMult_;
+bool transposed_; /*This is only used for the output layer which has a different layout from
+                   *all other matrices in the code. By default we shouldn't do an extra transpose.*/
 
-  PrepareBNodeOp(Expr input, Expr quant_mult, float clipValue)
-      : NaryNodeOp({input, quant_mult}, input->shape(), intgemm_<vtype>::intgemmType), clipValue_(clipValue) {
+  PrepareBNodeOp(Expr input, Expr quant_mult, float clipValue, bool transposed=false)
+      : NaryNodeOp({input, quant_mult}, newShape(input, transposed), intgemm_<vtype>::intgemmType), clipValue_(clipValue), transposed_(transposed) {
```

Comment:
@XapaJIaMnu Just for my understanding. `newShape(input, transposed)` will always store the shape of untransposed version of B matrix. Right?

Reply:
Yes, basically.